src/daemon/bitcoind/actions.rs

```diff
@@ -303,46 +304,54 @@ fn mark_confirmed_spends(
             .revault_unvault_txin(&unvault_descriptor, revaultd.read().unwrap().xpub_ctx());
         let unvault_outpoint = unvault_txin.outpoint();
 
-        let (spend_tx_hex, blockheight, _) = bitcoind
-            .get_wallet_transaction(&db_vault.spend_txid.expect("Must be set for 'spending'"))?;
-        if let Some(height) = blockheight {
-            db_mark_spent_unvault(&db_path, db_vault.id)?;
-            log::debug!(
-                "Spend tx '{}', spending Unvault '{}' was confirmed at height '{}'",
-                spend_tx_hex,
-                unvault_outpoint,
-                height
-            );
-        } else {
-            log::trace!(
-                "Spend tx '{}', spending Unvault '{}' is still unconfirmed",
-                spend_tx_hex,
-                unvault_outpoint
-            );
+        // First case: the tx might be spent. Note that if we don't have the spend_txid, of if
+        // get_wallet_transaction returns an Error, it's not a problem: then the tx was a cancel,
+        // and still is a cancel
+        // FIXME: maybe log error if db_vault.status is `Spending` but we don't have the
+        // spend_txid?
+        if let Some(spend_txid) = &db_vault.spend_txid {
+            if let Ok((spend_tx_hex, blockheight, _)) = bitcoind.get_wallet_transaction(spend_txid)
```

Comment:
I think it does not work as the transaction may still be in the wallet but not in the mempool. With the current code you could have a Cancel that gets RBF'ed by a Spend and you'll mark it as canceling i think.

Reply:
Actually I'm not sure you care if it's in the wallet or in the mempool. You just care about the blockheight, don't you?